### ISSUE TITLE
Force 240p support

### ISSUE THREAD

**Issue by GB-CG** [FIX94/Nintendont#162]
Not really an issue, but any chance of a 'Force 240p' option in the settings in the future?

**Comment by StateOfException** [comment]
Are there even any GameCube games that support 240p at all?

**Comment by FIX94** [comment]
well officially only very few like the mega man x collection use it, it can be technically patched in with some complicated video tricks as seen in swiss-gc, similar codes could be implemented in nintendont too, I just dont know when I'll get around to it.

**Comment by cmdreamer** [comment]
Sorry to revive this support request.
Now I understand why my LG 32" HD screen displays the "Invalid Format" error message while on the cut scenes when trying to play Megaman X Collection (NTSC) . Most of the time the game gets stuck there and I can't make it continue. I'm using a component video cable to play on my Red Wii with GC ports and a generic GC Joy (no issues with it).

An implementation of such functionality/option would fix this.

Does it exist an official list of such games? (ones that use 240p).

**Comment by GB-CG** [comment]
@cmdreamer Hm, interesting. I've found 240p in odd spots such as the video clips in Tony Hawk's 3.

**Comment by GB-CG** [comment]
@cmdreamer Odd question, Do you have a PS2 with component cables? Play a PS1 game with that, and if there's no picture, then your TV does not support 240p. Newer TV's tend to drop 240p support.

**Comment by cmdreamer** [comment]
@GB-CG Yes, thats what I meant, forcing 480p/i would solve the issues in case an HD TV drops support for 240p. And yes, I understand what you meant on your original post. A totally different approach to using 240p.

But what is more easy to implement and more useful to users:
-Forcing 480p/i instead of 240p, which might solve the "Invalid Format" issues, or
-Forcing 240p.

**Comment by GB-CG** [comment]
Ah ok.

**Comment by filevans** [comment]
yes this would be quite a good bonus feature

**Comment by danieljg** [comment]
Hey there, I just wanted to voice my opinion that this would be quite the nice option for Nintendon't to have.

To be specific, forcing 240p would be a great benefit to games like Nintendo Puzzle Collection, a game which looks substantially worse on a CRT than even the SNES original because of the interlaced output (unless you're in a 480p crt set.. but those aren't that common around here).

Also, I read that using the existing 'force progressive' option with a component out cable *without* turning on the progressive 480p option in the wii menu already works to give 240p output. I'm not sure if this is true but, although it sounds great, my crts take composite... so a proper 'force 240p' option would really work wonders.

**Comment by jaytea33** [comment]
Hello,

Just wanted to say +1 for forcing 240P mode, as this feature seems unique to Swiss and that doesn't work over USB.

I would really like to run Super Smash Bros. Melee in 240p mode rather than 480i, the progressive 240p signal appears much cleaner than even 480p in my opinion as seen here:

https://imgur.com/a/yies6

**Comment by Dockolm** [comment]
Hello there,

I'm also here to say it would be a nice option I would definitively use.
240p through OSSC x5 mode looks glorious on my HD screen, while 480p x2 isn't full screen, and stretching it + enable overscan isn't integer scaling thus looks okey but still kinda disappointing.

**Comment by matcarfer** [comment]
So, any news for 240p support on Nintendont?

**Comment by aarodynamics** [comment]
It would be great if I could fully switch over to Nintendont but I still use Swiss on my GameCube just so I can force 240p.

**Comment by bleggett** [comment]
RE2 and RE3 also benefit (imho) from being forced to 240p.

**Comment by filevans** [comment]
I think it would especially handy for forcing Gamecube Homebrew into 240p, like SQXZZ, not sure if that would be possible, does anyone know if nintendont can load gamecube homebrew? or if any wii program can do it?

**Comment by carnage702** [comment]
nintendont does not run any gc homebrew, nintendont works by patching sdk functions so games run, gc homebrew has no functions, everyone just codes what they want so they dont work.

**Comment by filevans** [comment]
> or if any wii program can do it?

**Comment by jaytea33** [comment]
Just thought I'd revisit this. As much as I'd like to see a 240p mode someday like Swiss has, it'd likely introduce issues for a while, and nintendont is currently stable and compatible.

These days with scaler options, seems wiser to take the 480p output from Nintendont and use a GBSControl or Retrotink, OSSC etc. to downscale to 240p if going to a CRT TV, and just passthrough 480p if going to a PC CRT.

However, it's the higher resolution scaling where it becomes a problem. if your max is 1200p on a PC CRT, and you start at 480p, that's an uneven 2.5 integer scale to get to 1200p. You're stuck with 960p if you 2x, and a 3x scale from 480p would exceed the monitor's max res. You'd need to start at 240p and 3x it to get to 720p or 5x it to get to 1200p properly, which means either use Swiss or 2 quality fast scalers - one to downscale to 240p, another to integer scale 3x (720p) or 5x (1200p)

However, 1440p is less of an issue. Starting at 480p, 3x it and call it a day - boom, you're at 1440p, and it's less image processing compared to starting at 240p and doing a 6x scale.

So right now, it's kind of an edge case issue for PC CRTs that max at 1200p, and you're kinda stuck with either native 480p or 960p. For that reason, I'd still like to see it in Nintendont someday, but that's a lot of work when better and more stable external hardware options are available, and 480p passthrough alone looks glorious as it is - you don't REALLY need 720p or 1200p for GC, but it's a fun novelty I hope to see someday in the future.

**Comment by abwezi** [comment]
@jaytea33 I just went with Swiss and an SD card adapter like was suggested to me. Only a handful of games really benefit from it anyways so a large SD is not needed I just keep all other backups on usb, and the sd memcard adapaters are cheap last I saw.

**Comment by smdstudios** [comment]
How does Swiss achieve 240p?
